src/hotspot/cpu/x86/peephole_x86.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+#include "precompiled.hpp"
+
+#ifdef COMPILER2
+
+#include "opto/peephole.hpp"
```

Comment:
I don't see why opto/peephole.hpp is useful.  Why not just include peephole_x86.hpp?  Then the empty peephole_<cpu>.hpp for the other platforms are no longer needed.

Reply:
How about including `peephole_<cpu>.hpp` only when "peepprocedure" is seen, and delete opto/peephole.hpp and empty `peephole_<cpu>.hpp` files?

Reply:
What do you think if I include `peephole_x86.hpp` in `x86_64.ad` in a source hpp block? This will result in the include appearing in `ad_x86.hpp`, which will be transitively included in `ad_x86_peephole.cpp`. Thanks a lot.

Reply:
Yes, good idea.

Reply:
Considering `peephole_x86.cpp` contains  only x64-specific code, I had a suggestion to rename it into `peephole_x86_64.cpp` and move `#ifdef _LP64` into `peephole_x86.hpp` to guard x64-specific declarations. But if you intend to include the header directly from `x86_64.ad`, you can rename the header to `peephole_x86_64.hpp` and get rid of `#ifdef _LP64` completely.

Reply:
I have refactored that and removed `opto/peephole.hpp` as well as other peephole headers. `peephole_x86.cpp` is also renamed to `peephole_x86_64.cpp`. Thanks very much.